From 84228d20feafdc5984c1550a6928d2b020db7052 Mon Sep 17 00:00:00 2001
From: John Audia <therealgraysky@proton.me>
Date: Fri, 11 Apr 2025 09:52:08 -0400
Subject: [PATCH] kernel: bump 6.6 to 6.6.87

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.87

Manually rebased:
	generic-hack/781-usb-net-rndis-support-asr.patch

All other patches automatically rebased.

Build system: x86/64
Build-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, x86/64
Run-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, x86/64

Signed-off-by: John Audia <therealgraysky@proton.me>
---
 include/kernel-6.6                            |  4 +--
 ...mstb-Add-DT-property-to-control-L1SS.patch |  2 +-
 ...0521-PCI-brcmstb-Add-BCM2712-support.patch | 34 +++++++++----------
 ...cmstb-Change-RCB_-MPS-64B-_MODE-bits.patch |  2 +-
 ...stb-optionally-extend-Tperst_clk-tim.patch |  2 +-
 ...pci-Disable-Host-Memory-Buffer-usage.patch |  6 ++--
 ...arn-if-no-host-bridge-NUMA-node-info.patch |  2 +-
 .../781-usb-net-rndis-support-asr.patch       | 10 +++---
 ...ng-with-source-address-failed-policy.patch | 22 ++++++------
 ...d-knob-for-filtering-rx-tx-BPDU-pack.patch |  2 +-
 ...IEI-vendor-prefix-and-IEI-WT61P803-P.patch |  2 +-
 11 files changed, 44 insertions(+), 44 deletions(-)

diff --git a/include/kernel-6.6 b/include/kernel-6.6
index 9fd98c30464c50..e3de08df98dee6 100644
--- a/include/kernel-6.6
+++ b/include/kernel-6.6
@@ -1,2 +1,2 @@
-LINUX_VERSION-6.6 = .86
-LINUX_KERNEL_HASH-6.6.86 = 49e3ad7423e40735faada0cd39665c071d47efd84ec3548acf119c9704f13e68
+LINUX_VERSION-6.6 = .87
+LINUX_KERNEL_HASH-6.6.87 = 8957e5c2dacdbc47a16dbf1f6303ca7088409be6197a3881f752313275357ac6
diff --git a/target/linux/bcm27xx/patches-6.6/950-0215-PCI-brcmstb-Add-DT-property-to-control-L1SS.patch b/target/linux/bcm27xx/patches-6.6/950-0215-PCI-brcmstb-Add-DT-property-to-control-L1SS.patch
index f29ecab0712880..c6a5810fade4d2 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0215-PCI-brcmstb-Add-DT-property-to-control-L1SS.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0215-PCI-brcmstb-Add-DT-property-to-control-L1SS.patch
@@ -65,7 +65,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  	writel(tmp, base + PCIE_MISC_HARD_PCIE_HARD_DEBUG);
  
  	return 0;
-@@ -1536,6 +1551,7 @@ static int brcm_pcie_probe(struct platfo
+@@ -1537,6 +1552,7 @@ static int brcm_pcie_probe(struct platfo
  	pcie->gen = (ret < 0) ? 0 : ret;
  
  	pcie->ssc = of_property_read_bool(np, "brcm,enable-ssc");
diff --git a/target/linux/bcm27xx/patches-6.6/950-0521-PCI-brcmstb-Add-BCM2712-support.patch b/target/linux/bcm27xx/patches-6.6/950-0521-PCI-brcmstb-Add-BCM2712-support.patch
index c4c79ae4281fd4..38f2e8631729e9 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0521-PCI-brcmstb-Add-BCM2712-support.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0521-PCI-brcmstb-Add-BCM2712-support.patch
@@ -831,7 +831,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  	return 0;
  }
  
-@@ -1207,6 +1534,7 @@ static void brcm_pcie_enter_l23(struct b
+@@ -1208,6 +1535,7 @@ static void brcm_pcie_enter_l23(struct b
  
  static int brcm_phy_cntl(struct brcm_pcie *pcie, const int start)
  {
@@ -839,7 +839,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  	static const u32 shifts[PCIE_DVT_PMU_PCIE_PHY_CTRL_DAST_NFLDS] = {
  		PCIE_DVT_PMU_PCIE_PHY_CTRL_DAST_PWRDN_SHIFT,
  		PCIE_DVT_PMU_PCIE_PHY_CTRL_DAST_RESET_SHIFT,
-@@ -1239,6 +1567,9 @@ static int brcm_phy_cntl(struct brcm_pci
+@@ -1240,6 +1568,9 @@ static int brcm_phy_cntl(struct brcm_pci
  		dev_err(pcie->dev, "failed to %s phy\n", (start ? "start" : "stop"));
  
  	return ret;
@@ -849,7 +849,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  }
  
  static inline int brcm_phy_start(struct brcm_pcie *pcie)
-@@ -1271,6 +1602,12 @@ static void brcm_pcie_turn_off(struct br
+@@ -1272,6 +1603,12 @@ static void brcm_pcie_turn_off(struct br
  	u32p_replace_bits(&tmp, 1, PCIE_MISC_HARD_PCIE_HARD_DEBUG_SERDES_IDDQ_MASK);
  	writel(tmp, base + PCIE_MISC_HARD_PCIE_HARD_DEBUG);
  
@@ -862,7 +862,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  	/* Shutdown PCIe bridge */
  	pcie->bridge_sw_init_set(pcie, 1);
  }
-@@ -1301,9 +1638,9 @@ static int brcm_pcie_suspend_noirq(struc
+@@ -1302,9 +1639,9 @@ static int brcm_pcie_suspend_noirq(struc
  	if (brcm_phy_stop(pcie))
  		dev_err(dev, "Could not stop phy for suspend\n");
  
@@ -874,7 +874,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  		return ret;
  	}
  
-@@ -1398,7 +1735,7 @@ err_regulator:
+@@ -1399,7 +1736,7 @@ err_regulator:
  	if (pcie->sr)
  		regulator_bulk_disable(pcie->sr->num_supplies, pcie->sr->supplies);
  err_reset:
@@ -883,7 +883,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  err_disable_clk:
  	clk_disable_unprepare(pcie->clk);
  	return ret;
-@@ -1410,8 +1747,8 @@ static void __brcm_pcie_remove(struct br
+@@ -1411,8 +1748,8 @@ static void __brcm_pcie_remove(struct br
  	brcm_pcie_turn_off(pcie);
  	if (brcm_phy_stop(pcie))
  		dev_err(pcie->dev, "Could not stop phy\n");
@@ -894,7 +894,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  	clk_disable_unprepare(pcie->clk);
  }
  
-@@ -1429,12 +1766,16 @@ static const int pcie_offsets[] = {
+@@ -1430,12 +1767,16 @@ static const int pcie_offsets[] = {
  	[RGR1_SW_INIT_1] = 0x9210,
  	[EXT_CFG_INDEX]  = 0x9000,
  	[EXT_CFG_DATA]   = 0x9004,
@@ -911,7 +911,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  };
  
  static const struct pcie_cfg_data generic_cfg = {
-@@ -1442,6 +1783,7 @@ static const struct pcie_cfg_data generi
+@@ -1443,6 +1784,7 @@ static const struct pcie_cfg_data generi
  	.type		= GENERIC,
  	.perst_set	= brcm_pcie_perst_set_generic,
  	.bridge_sw_init_set = brcm_pcie_bridge_sw_init_set_generic,
@@ -919,7 +919,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  };
  
  static const struct pcie_cfg_data bcm7425_cfg = {
-@@ -1449,6 +1791,7 @@ static const struct pcie_cfg_data bcm742
+@@ -1450,6 +1792,7 @@ static const struct pcie_cfg_data bcm742
  	.type		= BCM7425,
  	.perst_set	= brcm_pcie_perst_set_generic,
  	.bridge_sw_init_set = brcm_pcie_bridge_sw_init_set_generic,
@@ -927,7 +927,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  };
  
  static const struct pcie_cfg_data bcm7435_cfg = {
-@@ -1463,12 +1806,15 @@ static const struct pcie_cfg_data bcm490
+@@ -1464,12 +1807,15 @@ static const struct pcie_cfg_data bcm490
  	.type		= BCM4908,
  	.perst_set	= brcm_pcie_perst_set_4908,
  	.bridge_sw_init_set = brcm_pcie_bridge_sw_init_set_generic,
@@ -943,7 +943,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  };
  
  static const struct pcie_cfg_data bcm7278_cfg = {
-@@ -1476,6 +1822,7 @@ static const struct pcie_cfg_data bcm727
+@@ -1477,6 +1823,7 @@ static const struct pcie_cfg_data bcm727
  	.type		= BCM7278,
  	.perst_set	= brcm_pcie_perst_set_7278,
  	.bridge_sw_init_set = brcm_pcie_bridge_sw_init_set_7278,
@@ -951,7 +951,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  };
  
  static const struct pcie_cfg_data bcm2711_cfg = {
-@@ -1483,10 +1830,27 @@ static const struct pcie_cfg_data bcm271
+@@ -1484,10 +1831,27 @@ static const struct pcie_cfg_data bcm271
  	.type		= BCM2711,
  	.perst_set	= brcm_pcie_perst_set_generic,
  	.bridge_sw_init_set = brcm_pcie_bridge_sw_init_set_generic,
@@ -979,7 +979,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  	{ .compatible = "brcm,bcm4908-pcie", .data = &bcm4908_cfg },
  	{ .compatible = "brcm,bcm7211-pcie", .data = &generic_cfg },
  	{ .compatible = "brcm,bcm7278-pcie", .data = &bcm7278_cfg },
-@@ -1527,7 +1891,7 @@ static int brcm_pcie_probe(struct platfo
+@@ -1528,7 +1892,7 @@ static int brcm_pcie_probe(struct platfo
  
  	data = of_device_get_match_data(&pdev->dev);
  	if (!data) {
@@ -988,7 +988,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  		return -EINVAL;
  	}
  
-@@ -1538,6 +1902,7 @@ static int brcm_pcie_probe(struct platfo
+@@ -1539,6 +1903,7 @@ static int brcm_pcie_probe(struct platfo
  	pcie->type = data->type;
  	pcie->perst_set = data->perst_set;
  	pcie->bridge_sw_init_set = data->bridge_sw_init_set;
@@ -996,7 +996,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  
  	pcie->base = devm_platform_ioremap_resource(pdev, 0);
  	if (IS_ERR(pcie->base))
-@@ -1568,14 +1933,20 @@ static int brcm_pcie_probe(struct platfo
+@@ -1569,14 +1934,20 @@ static int brcm_pcie_probe(struct platfo
  		clk_disable_unprepare(pcie->clk);
  		return PTR_ERR(pcie->perst_reset);
  	}
@@ -1019,7 +1019,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  		clk_disable_unprepare(pcie->clk);
  		return ret;
  	}
-@@ -1598,6 +1969,33 @@ static int brcm_pcie_probe(struct platfo
+@@ -1599,6 +1970,33 @@ static int brcm_pcie_probe(struct platfo
  			dev_err(pcie->dev, "probe of internal MSI failed");
  			goto fail;
  		}
@@ -1053,7 +1053,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  	}
  
  	bridge->ops = pcie->type == BCM7425 ? &brcm7425_pcie_ops : &brcm_pcie_ops;
-@@ -1614,6 +2012,8 @@ static int brcm_pcie_probe(struct platfo
+@@ -1615,6 +2013,8 @@ static int brcm_pcie_probe(struct platfo
  		return ret;
  	}
  
diff --git a/target/linux/bcm27xx/patches-6.6/950-0695-PCI-brcmstb-Change-RCB_-MPS-64B-_MODE-bits.patch b/target/linux/bcm27xx/patches-6.6/950-0695-PCI-brcmstb-Change-RCB_-MPS-64B-_MODE-bits.patch
index dba20dc8544ae8..70d8229b4433ad 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0695-PCI-brcmstb-Change-RCB_-MPS-64B-_MODE-bits.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0695-PCI-brcmstb-Change-RCB_-MPS-64B-_MODE-bits.patch
@@ -45,7 +45,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  	writel(tmp, base + PCIE_MISC_MISC_CTRL);
  
  	brcm_pcie_set_tc_qos(pcie);
-@@ -1917,6 +1918,7 @@ static int brcm_pcie_probe(struct platfo
+@@ -1918,6 +1919,7 @@ static int brcm_pcie_probe(struct platfo
  
  	pcie->ssc = of_property_read_bool(np, "brcm,enable-ssc");
  	pcie->l1ss = of_property_read_bool(np, "brcm,enable-l1ss");
diff --git a/target/linux/bcm27xx/patches-6.6/950-0697-drivers-pci-brcmstb-optionally-extend-Tperst_clk-tim.patch b/target/linux/bcm27xx/patches-6.6/950-0697-drivers-pci-brcmstb-optionally-extend-Tperst_clk-tim.patch
index a0006d6a5a1403..604cf21b10ad3e 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0697-drivers-pci-brcmstb-optionally-extend-Tperst_clk-tim.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0697-drivers-pci-brcmstb-optionally-extend-Tperst_clk-tim.patch
@@ -61,7 +61,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  
  	/*
  	 * Wait for 100ms after PERST# deassertion; see PCIe CEM specification
-@@ -1919,6 +1940,7 @@ static int brcm_pcie_probe(struct platfo
+@@ -1920,6 +1941,7 @@ static int brcm_pcie_probe(struct platfo
  	pcie->ssc = of_property_read_bool(np, "brcm,enable-ssc");
  	pcie->l1ss = of_property_read_bool(np, "brcm,enable-l1ss");
  	pcie->rcb_mps_mode = of_property_read_bool(np, "brcm,enable-mps-rcb");
diff --git a/target/linux/bcm27xx/patches-6.6/950-1438-nvme-pci-Disable-Host-Memory-Buffer-usage.patch b/target/linux/bcm27xx/patches-6.6/950-1438-nvme-pci-Disable-Host-Memory-Buffer-usage.patch
index 4f1ea9249c65a8..02c4dc449899e4 100644
--- a/target/linux/bcm27xx/patches-6.6/950-1438-nvme-pci-Disable-Host-Memory-Buffer-usage.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-1438-nvme-pci-Disable-Host-Memory-Buffer-usage.patch
@@ -18,7 +18,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
 
 --- a/drivers/nvme/host/pci.c
 +++ b/drivers/nvme/host/pci.c
-@@ -1932,6 +1932,7 @@ static void nvme_free_host_mem(struct nv
+@@ -1948,6 +1948,7 @@ static void nvme_free_host_mem(struct nv
  	dev->nr_host_mem_descs = 0;
  }
  
@@ -26,7 +26,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  static int __nvme_alloc_host_mem(struct nvme_dev *dev, u64 preferred,
  		u32 chunk_size)
  {
-@@ -2000,9 +2001,11 @@ out:
+@@ -2016,9 +2017,11 @@ out:
  	dev->host_mem_descs = NULL;
  	return -ENOMEM;
  }
@@ -38,7 +38,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  	u64 min_chunk = min_t(u64, preferred, PAGE_SIZE * MAX_ORDER_NR_PAGES);
  	u64 hmminds = max_t(u32, dev->ctrl.hmminds * 4096, PAGE_SIZE * 2);
  	u64 chunk_size;
-@@ -2015,6 +2018,7 @@ static int nvme_alloc_host_mem(struct nv
+@@ -2031,6 +2034,7 @@ static int nvme_alloc_host_mem(struct nv
  			nvme_free_host_mem(dev);
  		}
  	}
diff --git a/target/linux/bcm27xx/patches-6.6/950-1452-Revert-PCI-Warn-if-no-host-bridge-NUMA-node-info.patch b/target/linux/bcm27xx/patches-6.6/950-1452-Revert-PCI-Warn-if-no-host-bridge-NUMA-node-info.patch
index 5aa88661c3ea05..470a90cb97efde 100644
--- a/target/linux/bcm27xx/patches-6.6/950-1452-Revert-PCI-Warn-if-no-host-bridge-NUMA-node-info.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-1452-Revert-PCI-Warn-if-no-host-bridge-NUMA-node-info.patch
@@ -17,7 +17,7 @@ Signed-off-by: Dom Cobley <popcornmix@gmail.com>
 
 --- a/drivers/pci/probe.c
 +++ b/drivers/pci/probe.c
-@@ -968,9 +968,6 @@ static int pci_register_host_bridge(stru
+@@ -967,9 +967,6 @@ static int pci_register_host_bridge(stru
  	else
  		pr_info("PCI host bridge to bus %s\n", name);
  
diff --git a/target/linux/generic/hack-6.6/781-usb-net-rndis-support-asr.patch b/target/linux/generic/hack-6.6/781-usb-net-rndis-support-asr.patch
index 47339b6c22f7b2..d5cc7e0a7ef3ef 100644
--- a/target/linux/generic/hack-6.6/781-usb-net-rndis-support-asr.patch
+++ b/target/linux/generic/hack-6.6/781-usb-net-rndis-support-asr.patch
@@ -13,7 +13,7 @@ Signed-off-by: Daniel Golle <daniel@makrotopia.org>
 
 --- a/drivers/net/usb/rndis_host.c
 +++ b/drivers/net/usb/rndis_host.c
-@@ -630,6 +630,16 @@ static const struct driver_info	zte_rndi
+@@ -640,6 +640,16 @@ static const struct driver_info	wwan_rnd
  	.tx_fixup =	rndis_tx_fixup,
  };
  
@@ -30,7 +30,7 @@ Signed-off-by: Daniel Golle <daniel@makrotopia.org>
  /*-------------------------------------------------------------------------*/
  
  static const struct usb_device_id	products [] = {
-@@ -666,6 +676,36 @@ static const struct usb_device_id	produc
+@@ -676,6 +686,36 @@ static const struct usb_device_id	produc
  	USB_INTERFACE_INFO(USB_CLASS_WIRELESS_CONTROLLER, 1, 3),
  	.driver_info = (unsigned long) &rndis_info,
  }, {
@@ -64,6 +64,6 @@ Signed-off-by: Daniel Golle <daniel@makrotopia.org>
 +				      USB_CLASS_WIRELESS_CONTROLLER, 1, 3),
 +	.driver_info = (unsigned long) &asr_rndis_info,
 +}, {
- 	/* Novatel Verizon USB730L */
- 	USB_INTERFACE_INFO(USB_CLASS_MISC, 4, 1),
- 	.driver_info = (unsigned long) &rndis_info,
+ 	/* Mobile Broadband Modem, seen in Novatel Verizon USB730L and
+ 	 * Telit FN990A (RNDIS)
+ 	 */
diff --git a/target/linux/generic/pending-6.6/670-ipv6-allow-rejecting-with-source-address-failed-policy.patch b/target/linux/generic/pending-6.6/670-ipv6-allow-rejecting-with-source-address-failed-policy.patch
index ccafdf291136ee..abd6586e6cb2bb 100644
--- a/target/linux/generic/pending-6.6/670-ipv6-allow-rejecting-with-source-address-failed-policy.patch
+++ b/target/linux/generic/pending-6.6/670-ipv6-allow-rejecting-with-source-address-failed-policy.patch
@@ -138,7 +138,7 @@ Signed-off-by: Jonas Gorski <jogo@openwrt.org>
  static const struct rt6_info ip6_blk_hole_entry_template = {
  	.dst = {
  		.__rcuref	= RCUREF_INIT(1),
-@@ -1043,6 +1057,7 @@ static const int fib6_prop[RTN_MAX + 1]
+@@ -1077,6 +1091,7 @@ static const int fib6_prop[RTN_MAX + 1]
  	[RTN_BLACKHOLE]	= -EINVAL,
  	[RTN_UNREACHABLE] = -EHOSTUNREACH,
  	[RTN_PROHIBIT]	= -EACCES,
@@ -146,7 +146,7 @@ Signed-off-by: Jonas Gorski <jogo@openwrt.org>
  	[RTN_THROW]	= -EAGAIN,
  	[RTN_NAT]	= -EINVAL,
  	[RTN_XRESOLVE]	= -EINVAL,
-@@ -1078,6 +1093,10 @@ static void ip6_rt_init_dst_reject(struc
+@@ -1112,6 +1127,10 @@ static void ip6_rt_init_dst_reject(struc
  		rt->dst.output = ip6_pkt_prohibit_out;
  		rt->dst.input = ip6_pkt_prohibit;
  		break;
@@ -157,7 +157,7 @@ Signed-off-by: Jonas Gorski <jogo@openwrt.org>
  	case RTN_THROW:
  	case RTN_UNREACHABLE:
  	default:
-@@ -4554,6 +4573,17 @@ static int ip6_pkt_prohibit_out(struct n
+@@ -4588,6 +4607,17 @@ static int ip6_pkt_prohibit_out(struct n
  	return ip6_pkt_drop(skb, ICMPV6_ADM_PROHIBITED, IPSTATS_MIB_OUTNOROUTES);
  }
  
@@ -175,7 +175,7 @@ Signed-off-by: Jonas Gorski <jogo@openwrt.org>
  /*
   *	Allocate a dst for local (unicast / anycast) address.
   */
-@@ -5045,7 +5075,8 @@ static int rtm_to_fib6_config(struct sk_
+@@ -5079,7 +5109,8 @@ static int rtm_to_fib6_config(struct sk_
  	if (rtm->rtm_type == RTN_UNREACHABLE ||
  	    rtm->rtm_type == RTN_BLACKHOLE ||
  	    rtm->rtm_type == RTN_PROHIBIT ||
@@ -185,7 +185,7 @@ Signed-off-by: Jonas Gorski <jogo@openwrt.org>
  		cfg->fc_flags |= RTF_REJECT;
  
  	if (rtm->rtm_type == RTN_LOCAL)
-@@ -6307,6 +6338,8 @@ static int ip6_route_dev_notify(struct n
+@@ -6341,6 +6372,8 @@ static int ip6_route_dev_notify(struct n
  #ifdef CONFIG_IPV6_MULTIPLE_TABLES
  		net->ipv6.ip6_prohibit_entry->dst.dev = dev;
  		net->ipv6.ip6_prohibit_entry->rt6i_idev = in6_dev_get(dev);
@@ -194,7 +194,7 @@ Signed-off-by: Jonas Gorski <jogo@openwrt.org>
  		net->ipv6.ip6_blk_hole_entry->dst.dev = dev;
  		net->ipv6.ip6_blk_hole_entry->rt6i_idev = in6_dev_get(dev);
  #endif
-@@ -6318,6 +6351,7 @@ static int ip6_route_dev_notify(struct n
+@@ -6352,6 +6385,7 @@ static int ip6_route_dev_notify(struct n
  		in6_dev_put_clear(&net->ipv6.ip6_null_entry->rt6i_idev);
  #ifdef CONFIG_IPV6_MULTIPLE_TABLES
  		in6_dev_put_clear(&net->ipv6.ip6_prohibit_entry->rt6i_idev);
@@ -202,7 +202,7 @@ Signed-off-by: Jonas Gorski <jogo@openwrt.org>
  		in6_dev_put_clear(&net->ipv6.ip6_blk_hole_entry->rt6i_idev);
  #endif
  	}
-@@ -6518,6 +6552,8 @@ static int __net_init ip6_route_net_init
+@@ -6552,6 +6586,8 @@ static int __net_init ip6_route_net_init
  
  #ifdef CONFIG_IPV6_MULTIPLE_TABLES
  	net->ipv6.fib6_has_custom_rules = false;
@@ -211,7 +211,7 @@ Signed-off-by: Jonas Gorski <jogo@openwrt.org>
  	net->ipv6.ip6_prohibit_entry = kmemdup(&ip6_prohibit_entry_template,
  					       sizeof(*net->ipv6.ip6_prohibit_entry),
  					       GFP_KERNEL);
-@@ -6528,11 +6564,21 @@ static int __net_init ip6_route_net_init
+@@ -6562,11 +6598,21 @@ static int __net_init ip6_route_net_init
  			 ip6_template_metrics, true);
  	INIT_LIST_HEAD(&net->ipv6.ip6_prohibit_entry->dst.rt_uncached);
  
@@ -234,7 +234,7 @@ Signed-off-by: Jonas Gorski <jogo@openwrt.org>
  	net->ipv6.ip6_blk_hole_entry->dst.ops = &net->ipv6.ip6_dst_ops;
  	dst_init_metrics(&net->ipv6.ip6_blk_hole_entry->dst,
  			 ip6_template_metrics, true);
-@@ -6559,6 +6605,8 @@ out:
+@@ -6593,6 +6639,8 @@ out:
  	return ret;
  
  #ifdef CONFIG_IPV6_MULTIPLE_TABLES
@@ -243,7 +243,7 @@ Signed-off-by: Jonas Gorski <jogo@openwrt.org>
  out_ip6_prohibit_entry:
  	kfree(net->ipv6.ip6_prohibit_entry);
  out_ip6_null_entry:
-@@ -6578,6 +6626,7 @@ static void __net_exit ip6_route_net_exi
+@@ -6612,6 +6660,7 @@ static void __net_exit ip6_route_net_exi
  	kfree(net->ipv6.ip6_null_entry);
  #ifdef CONFIG_IPV6_MULTIPLE_TABLES
  	kfree(net->ipv6.ip6_prohibit_entry);
@@ -251,7 +251,7 @@ Signed-off-by: Jonas Gorski <jogo@openwrt.org>
  	kfree(net->ipv6.ip6_blk_hole_entry);
  #endif
  	dst_entries_destroy(&net->ipv6.ip6_dst_ops);
-@@ -6661,6 +6710,9 @@ void __init ip6_route_init_special_entri
+@@ -6695,6 +6744,9 @@ void __init ip6_route_init_special_entri
  	init_net.ipv6.ip6_prohibit_entry->rt6i_idev = in6_dev_get(init_net.loopback_dev);
  	init_net.ipv6.ip6_blk_hole_entry->dst.dev = init_net.loopback_dev;
  	init_net.ipv6.ip6_blk_hole_entry->rt6i_idev = in6_dev_get(init_net.loopback_dev);
diff --git a/target/linux/generic/pending-6.6/710-bridge-add-knob-for-filtering-rx-tx-BPDU-pack.patch b/target/linux/generic/pending-6.6/710-bridge-add-knob-for-filtering-rx-tx-BPDU-pack.patch
index 31e29c447e8ade..d62ccc571ce898 100644
--- a/target/linux/generic/pending-6.6/710-bridge-add-knob-for-filtering-rx-tx-BPDU-pack.patch
+++ b/target/linux/generic/pending-6.6/710-bridge-add-knob-for-filtering-rx-tx-BPDU-pack.patch
@@ -161,7 +161,7 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
  
  struct rtnl_link {
  	rtnl_doit_func		doit;
-@@ -4978,7 +4978,9 @@ int ndo_dflt_bridge_getlink(struct sk_bu
+@@ -4981,7 +4981,9 @@ int ndo_dflt_bridge_getlink(struct sk_bu
  	    brport_nla_put_flag(skb, flags, mask,
  				IFLA_BRPORT_MCAST_FLOOD, BR_MCAST_FLOOD) ||
  	    brport_nla_put_flag(skb, flags, mask,
diff --git a/target/linux/mvebu/patches-6.6/901-dt-bindings-Add-IEI-vendor-prefix-and-IEI-WT61P803-P.patch b/target/linux/mvebu/patches-6.6/901-dt-bindings-Add-IEI-vendor-prefix-and-IEI-WT61P803-P.patch
index e9c1d57d244e12..2f7361c93edb68 100644
--- a/target/linux/mvebu/patches-6.6/901-dt-bindings-Add-IEI-vendor-prefix-and-IEI-WT61P803-P.patch
+++ b/target/linux/mvebu/patches-6.6/901-dt-bindings-Add-IEI-vendor-prefix-and-IEI-WT61P803-P.patch
@@ -207,7 +207,7 @@ Cc: Robert Marko <robert.marko@sartura.hr>
 +    };
 --- a/Documentation/devicetree/bindings/vendor-prefixes.yaml
 +++ b/Documentation/devicetree/bindings/vendor-prefixes.yaml
-@@ -609,6 +609,8 @@ patternProperties:
+@@ -611,6 +611,8 @@ patternProperties:
      description: IC Plus Corp.
    "^idt,.*":
      description: Integrated Device Technologies, Inc.
